lib/api/v3/work_packages/schema/work_package_schema_representer.rb

```diff
@@ -0,0 +1,251 @@
+#-- encoding: UTF-8
+#-- copyright
+# OpenProject is a project management system.
+# Copyright (C) 2012-2015 the OpenProject Foundation (OPF)
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See doc/COPYRIGHT.rdoc for more details.
+#++
+
+require 'roar/decorator'
+require 'roar/json/hal'
+
+module API
+  module V3
+    module WorkPackages
+      module Schema
+        class WorkPackageSchemaRepresenter < ::API::Decorators::Single
+          I18N_WORK_PACKAGE = 'activerecord.attributes.work_package'
+
+          def self.schema(property,
+                          type: nil,
+                          title: nil,
+                          required: true,
+                          writable: true,
+                          min_length: nil,
+                          max_length: nil)
+            raise ArgumentError if property.nil? || type.nil?
+
+            title = I18n.t("#{I18N_WORK_PACKAGE}.#{property}") unless title
+
+            schema = ::API::Decorators::PropertySchemaRepresenter.new(type: type,
+                                                                      name: title)
+            schema.required = required
+            schema.writable = writable
+            schema.min_length = min_length if min_length
+            schema.max_length = max_length if max_length
+
+            property property,
+                     getter: -> (*) { schema },
+                     writeable: false
+          end
+
+          def self.schema_with_allowed_link(property,
```

Comment:
I will not address this here, because in a follow up branch this method was moved out and apparently made even worse... I will need to fix it there...